Tomb of Rafid: The Away Mission

unid15 opened the pull request:
![](http://puu.sh/p3X97/27b9c6cb81.jpg)

A long time ago, in a galaxy far away, there was a planet under the rule of a man named Rafid the Mad. Eventually he died and his people built a great pyramid to bury him in. Today, that civilization no longer exists - but the great pyramid still stands. A few months ago team of archaeologists set up a gateway and teleported there to attempt to unlock the tomb. We haven't heard from them since.

That's where you come in, crew of SS13. Your gateway has been connected to the (now empty) expedition camp. It's your duty to find out where the archaeologists have disappeared. Feel free to loot the shit out of the tomb, too

SPOILERS BELOW

---

This away mission features several new monsters:
- Mummy: slow but tough monster. On death, releases a cloud of miasma that deals toxin damage and knocks out any nearby enemies that breathe it in. Mummy rags can be taken off their bodies and worn as a clothing item!
- Mummy warrior: more powerful than normal mummies, they wield a spear and a buckler
- Mummy priest: fragile but more dangerous than a normal mummy. Can cast a curse at his enemies that can blind them, deafen them, cause brain damage/hallucinations or deal damage. Also wields a 
  knife which lets it defend itself from foes that come in too close
- Mummy high priest: a boss found in a side branch. His curse is much more powerful. He can also cast polymorph (read: shoot a staff of change projectile) to mutate his enemies
- Flying skull: like viscerators, but spookier. Most of them are sleeping, so be careful not to wake them up (by stepping on them or picking them up)
- Jackal: an undead creature with the appearance of a human with a jackal's head. Fast, but not very dangerous
- Jackal Firebreather: a jackal with the ability to breathe fire
- Jackal Embalmer: a jackal with the ability to raise dead corpses as mummies. If a corpse is controlled by a player, the player becomes a mummy. Dead mummies can be resurrected too, effectively making mummies immortal as long as there's an embalmer nearby. The ritual isn't instant and takes some time to perform. Embalmers don't engage in fighting and ignore their foes in favor of resurrecting fallen allies

As well as old monsters:
- Mimics, space bats, spiders, spider queen, wizard

**LOCATIONS**
- Great Pyramid - contains the entrance to the Tomb of Rafid. The entrance is sealed; it will only unlock after somebody sacrifices Ian or a living human (braindeads and monkey humans don't count) on the sacrificial altar in front of it
- Tomb of Rafid - main dungeon. Those who get to the end will find 50 diamonds, 50 phazon, a cult blade, a happy mask and a staff of necromancy. There is no boss, but there are many mummies and jackals on the way to the burial site
- Tower of Madness - side branch consisting of 6 levels. The levels contain traps and dangerous monsters. Those who get to the end and defeat the High Priest of Riniel will find a lifetime supply of cursed d20s.
- Spider Caverns - side branch. Those who defeat the spider queen and get to the end will find a wizard's den with a staff of animation and an angry wizard inside.

**GIFS OF NEW MONSTERS IN ACTION**

Mummy: http://puu.sh/p4A6b/1bc3fe3775.gif

Mummy priest: http://puu.sh/p4AeD/308a272a39.gif

Jackal Embalmer: http://puu.sh/p4A8O/d896e136c0.gif

Jackal Firebreather: http://puu.sh/p4AaF/98bc1bf044.gif

**MAP OF THE DUNGEON**

http://puu.sh/p4AlW/3a4d1b9f34.jpg

icantthinkofanameritenow commented:
Cult blades aren't that useful if you're not a cultist since they fuck with your vision and all. Maybe replace it with that Catechizer mace or something?

Sacrificing a live human or Ian to get in may be an issue, though; perhaps allow any sacrifice to open it up, since even Nar-Nar is fine with having monkeys sacrificed to him and it won't take people out of the game unnecessarily.

unid15 commented:
Maybe turn the sacrificed human into a monster somewhere in the dungeon. What do you think?

icantthinkofanameritenow commented:
@unid15
A previously sacrificed victim or the one used as a sacrifice by the crew? The former would make a bit more sense, since it could be fluffed out as one of the archaeologists taking desperate measures to open up the tomb proper or something. 

But if it's the latter, I would strongly advise finding a way to allow whoever was controlling the sacrificed victim to retain control over the newly created mob, since otherwise you're still permanently taking him out of the round with no benefit to him. Especially since if the body is converted to mob, you couldn't just clone him after using him as a sacrifice as I speculated might be theoretically possible.

Imagebard commented:
Since you're here, can you add a "mummy" setting to the staff of necromancy?

icantthinkofanameritenow commented:
Also, for the wizard den why not add one of the one-use spell books?  There's a fairly decent selection available that might be good loot- if memory serves, there's Fireball, Smoke, Forcewall, Blind, Knock. Mind Swap (though that one works a bit differently according to the code and I don't know if it works as it should), and Curse of the Horseman.There's also another one that doesn't seem to correspond to a normal wizard spell, but using that probably won't be a good idea. 

I'd personally add one of the more innocuous ones like Forcewall, but that's just me.

Last, as a boss for the main Tomb of Rafid, why not add Rafid himself as a boss? I'd figure it should be simple enough to make him a named high priest with more health and maybe a fancy crown.

icantthinkofanameritenow commented on code/modules/mob/living/simple_animal/hostile/human/mummy.dm:
```diff
@@ -0,0 +1,221 @@
+		cardinals = c
+
+		chemholder.reagents.add_reagent("toxin", 50)
+		chemholder.reagents.add_reagent("cyanide", 20)
```
You do know cyanide at that dosage is extremely deadly, right? If I were you, I'd replace it with sleep toxin or even 5-8u of chloral hydrate.

Kurfursten replied:
Cyanide is deadly at 3u

unid15 replied:
It's spread out over multiple clouds, so it's not THAT bad (see the mummy gif, the same dosage is used there). I'll still lower it a bit though

Kurfursten commented:
More than anything, the sacrifice requirement reminds me of one shift that an antag uploaded a law that required "someone must be sacrificed every 5 minutes", so one guy volunteered to stay in medbay being sacrificed every five minutes and revived.

unid15 commented:
@Imagebard the staff of necromancy has some low quality code and probably needs to be reworked

I'll do it some time later

SkowronX commented:
necromancer staff and undead mobs really need an update. Especially for player controlled ones.

icantthinkofanameritenow commented:
@Kurfursten 

In this case the body is immediately destroyed as if a soul stone shard was used on it, so the comparison isn't entirely accurate. However, that creates even bigger problems since the sacrificed person can't come back to life through the most common methods (e.g. cloning, defibrillator). And while Ian can be sacrificed instead of a crew member, I don't see many people who would be willing to do it- and in that rule revision Nuke is working on now, killing someone's pet makes you valid to them, which needless to say could cause a lot of trouble. 

I quote:

>The direct owner of a pet may retaliate, up to killing, a murderer of their pet, accidents notwithstanding.

clusterfack commented:
Exciting

clusterfack commented on code/game/turfs/simulated/mineral.dm:
```diff
@@ -11,12 +11,8 @@
 /turf/simulated/floor/mineral/plasma
 	name = "plasma floor"
 	icon_state = "plasma"
-	floor_tile = /obj/item/stack/tile/mineral/plasma
```
I know I should ask why, so I'll just ask. Why?

Also wow I never realized how bad this system was before.

unid15 replied:
There used to be a returnToPool(floor_tile) below that caused a lot of runtimes

clusterfack replied:
What was the runtime?

unid15 replied:
The runtime that happens when you do " returnToPool(/obj/item/stack/tile/mineral/plasma) "

clusterfack replied:
I'm guessing the reason I haven't seen it is that this file is unticked?

unid15 replied:
Yep, the file was unticked and these turfs weren't used on any map

clusterfack replied:
I feel like the New() call for all these should just be getFromPool(floor_tile, null) at the base level since every single variation of this New() is doing the same thing

FudgePucker commented:
Having something that demanding to enter will make it more unique instead of being another loot grab.

Also do genetic back ups work?

unid15 commented:
They should work, yeah. I'm not sure if I should disable that or let experienced players fool the ruler of the underworld this way

icantthinkofanameritenow commented:
@unid15
Well if you put them in a mob afterwards it obviously won't work, and a god of death and the underworld is not going to take an attempt to swindle him lightly, let alone be actually fooled by it. (I'm not entirely sure if backups even work right now, actually. They didn't seem to be working last I checked. )

Still, there's that big issue- reversible or not, I believe that nobody is going to sacrifice themselves for loot that by definition they will be unable to get, and if it's not done voluntarily then it's flat-out murder (possibly as a nonantag, which will give the admins a headache). It's going to be a massive BWOINK magnet at the best of times, and it doesn't take a genius to see that if any Asimov silicons find out about that, nobody's going to be getting in that pyramid since human sacrifice is blatantly harmful to humans. 

Admins who are reading this, what's your take on it? It comes off to me as being one of those things that sounds cool on paper, but ends up getting tied up in dozens of practical difficulties when it's actually implemented.

DeityLink commented:
I volunteer to use my spriting skills to add some details/overlays in there if you wish.

unid15 commented:
Would be great @DeityLink

Arthurdentist commented:
@unid15 
I'm about to make a PR which will add all the missing single use spellbooks to the game. If you wanted you could use some of these as loot.

unid15 commented:
I'll be away next week so I won't be able to work on this for a few days

New updates (spoiler alert):

http://puu.sh/p7nJk/5d0d9a1db1.jpg

http://puu.sh/p7nRp/4ed577e103.png

* Separated the main hall into smaller rooms

* Placed some oil spills to make fighting firebreathers more fun

* Added fire traps (moving structures that shoot fire in a straight line every set interval)

* Added jackal pyromancer to one of the rooms (like firebreather but shoots 3x as much fire - harder to dodge)

* Added a challenge at the beginning of the tomb where you have to run through narrow hallways that are being periodically blasted by fire traps. At the end there's a button that opens the gate to the rest of the tomb. In front of the gate there are wooden support beams - setting them on fire will collapse them, permanently blocking the gate off

* There's another way to go through that gate - a dark passage full of jackals

* The final chamber before the burial vault now contains a big gate. You have to press 2 buttons in nearby rooms to open the gate

* Added a pretty secret room with a horsehead single use spellbook

technical stuff:

* Added "secret doors" for mapping. They are like invisible landmarks that change the turf under them into a floor (and then restore the original turf when triggered again). Also added buttons that trigger secret doors.

clusterfack commented:
Is there a spiky pit trap yet? And an indiana jones boulder?

Arthurdentist commented:
https://www.youtube.com/watch?v=KFREs4GaJVU

unid15 commented:
Can we do a few test runs on the server to gather some general feedback while I'm gone?

clusterfack commented:
I don't see anything wrong with this codewise, and since it is an away mission it wont break anything if it doesn't 100% operate correctly so we'll test merge it.

clusterfack commented on code/game/objects/effects/traps.dm:
```diff
@@ -0,0 +1,28 @@
+/obj/effect/trap
```
I refactored mines to be extremely similar to this code unknowingly, one of them should probably be brought under the other's typepath.

DeityLink commented:
Didn't have time to improve it after all. I've been crazy busy this week. oh well.

Shadowmech88 commented:
Can always just make another PR for the sprite improvements if you want to.
